src/wifi.rs

```diff
@@ -1684,22 +1735,27 @@ where
         )
     }
 
+    /// As per [`WifiDriver::disconnect()`]
     pub fn disconnect(&mut self) -> Result<(), EspError> {
         self.wifi.disconnect()?;
         self.wifi_wait_while(|| self.wifi.is_connected(), None)
     }
 
+    /// As per [`WifiDriver::scan_n()`]
     pub fn scan_n<const N: usize>(
         &mut self,
     ) -> Result<(heapless::Vec<AccessPointInfo, N>, usize), EspError> {
         self.wifi.scan_n()
     }
 
+    /// As per [`WifiDriver::scan()`]
     #[cfg(feature = "alloc")]
     pub fn scan(&mut self) -> Result<alloc::vec::Vec<AccessPointInfo>, EspError> {
         self.wifi.scan()
     }
 
+    /// Waits for [`WifiEvent`]s. The wait ends when a wifi event happens, and
```

Comment:
Correction: not for that event. The function provided by the user does not have access to the actual event.
Perhaps a more correct wording would be:
```
Waits for a certain condition provided by the user in the form of a `matcher` callback to become true. Most often than not that condition would be related to the state of the Wifi driver. In other words, whether the driver is started, stopped, (dis)connected and so on.

Note that the waiting is not done internally using busy-looping and/or timeouts. Rather, the condition (`matcher`) is evaluated initially, and if it returns `false`, it is re-evaluated each time the ESP IDF C 
Wifi driver posts a Wifi event on the system event loop. The reasoning behind this is that changes to the state of the Wifi driver are always accompanied by posting Wifi events.  
```